db/migrate/20240115144930_create_tokens.rb

```diff
@@ -16,7 +16,7 @@ def change
       t.index [:deploy_block_number, :deploy_transaction_index], unique: true
       
       t.check_constraint "protocol ~ '^[a-z0-9\-]{1,18}$'"
-      t.check_constraint "tick ~ '^[a-zA-Z0-9]{1,18}$'"
+      t.check_constraint "tick ~ '^[[:alnum:]\p{Emoji_Presentation}]{1,18}$'"
```

Comment:
Wow I had never heard of `Emoji_Presentation`! Very nice. Though now I wonder: should we remove all restrictions? Token indexing is "opt-in" and so if the person running the indexer wants to do something crazy then maybe we let them?

Also I need to check whether I'm sufficiently sanitizing these: https://github.com/0xFacet/ethscriptions-indexer/pull/14/files#diff-1607c83626dc7b2d0572e7d07e8f1fd859bf55b0334af345e2a5f9e344ece109R45

Reply:
Thank you for adopting this suggestion; your points are very valid.
I also noticed some potential issues when people deploy and mint tokens, and I believe certain constraints should be imposed. For example, there should be constraints on the maximum length of tick names and the maximum value of total_supply.

Without some limitations, the following problems may arise, as we have encountered:

1. In terms of database design, there could be performance losses due to setting field lengths excessively large for a very small number of ticks or total_supply.
2. In the design of apps and websites, additional effort is needed to accommodate them because they are excessively long and may exceed screen limits.
3. The existence of these outlier values might complicate the development of applications based on the indexer.

Here are some instances where I found abnormal data in the inscriptions:

1. Excessively large total_supply values:
   - ethscription_number: 1045075, eacc, 210000000000000000000000000000000000000000
   - ethscription_number: 895511, billion, 115132219018763992565095597973971522400
   - ethscription_number: 3133016, ethk, 486418694186186186168186186186161600000
   - ethscription_number: 1445420, maxi, 210000000000000000000000000000
   - ethscription_number: 1233125, EMD, 210000000000000000000000000000
   - ethscription_number: 2688766, args, 1000000000000000000000

2. Abnormal tick values:
   - ethscription_number: 3532421, data:,{"p":"erc-20","op":"deploy","tick":"\"Akita inu\"","max":"21000000","lim":"1000"}

3. Abnormal mint id:
   - ethscription_number: 35078, mint_id: 20315.1

Finally, thank you for open-sourcing the indexer, making the development of applications based on inscriptions simpler. Our website ([ethinscription.org](https://ethinscription.org)) has already been built using data from ethscriptions-indexer, and we look forward to the token and domain indexing features of ethscriptions-indexer.

Reply:
Interesting!

But the more I think about it the more I think we should leave it to indexer users and just set some broad guardrails on length (which I have as 1000) to make sure the database doesn't get bogged down.

Beyond that I think users should be able to use whatever standard they want or define their own; I don't think the base protocol should take a stance on valid tokens. It should be a higher-level protocol which this indexer should support but not pick. Instead I think this indexer should attempt to organize and classify what people are already doing so that we will support everything popular versus having to exclude popular tokens because they don't fit a certain standard.

If we do want standards and interoperability we can use actual ERC20 in Facet!

Reply:
This is also a great idea, allowing more people to use indexer. Next, I will explore FacetVM.😊